src/Download.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react'
+import axios from 'axios'
+import fileDownload from 'js-file-download'
+
+export default function Download({ query }) {
+  const [selected, setSelected] = useState('json')
+  const options = ['json', 'csv', 'xlsx', 'tsv', 'ods']
+  const [downloading, setDownloading] = useState('')
+
+  const handleChange = (event) => {
+    setSelected(event.target.value)
+  }
+
+  const downloadData = () => {
+    setDownloading('Preparing Download')
+    axios(`http://localhost:8080/v1/download?format=${selected}`, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      responseType: 'blob',
+      data: {
+        query: `query Dataset {
+                    ${query}
+                    }
+                  `,
+      },
+    })
+      .then((response) => {
+        setDownloading('Downloading file')
+        fileDownload(response.data, `data.${selected}`)
+        setDownloading('Done.')
+        setTimeout(() => setDownloading(''), 5000)
```

Comment:
Not sure what is this...

Reply:
That was to update the message and after the message is done, after 5 seconds the message disappears. But I will change on the spinner